core/src/main/java/org/apache/iceberg/SerializableTable.java

```diff
@@ -107,7 +108,13 @@ private String metadataFileLocation(Table table) {
 
   private FileIO fileIO(Table table) {
     if (table.io() instanceof HadoopConfigurable) {
-      ((HadoopConfigurable) table.io()).serializeConfWith(conf -> new SerializableConfiguration(conf)::get);
+      HadoopConfigurable configurableIo = (HadoopConfigurable) table.io();
+      if ("true".equalsIgnoreCase(configurableIo.getConf().get(ConfigProperties.CONFIG_SERIALIZATION_DISABLED))) {
+        // serialize with an empty Configuration object - configs should be populated on the deserializer-side
+        configurableIo.serializeConfWith(conf -> new SerializableConfiguration(new Configuration(false))::get);
```

Comment:
Shouldn't we create a new implementation without the extra magic / code / method calls in `SerializableConfiguration`?

Reply:
Good point. Created a new implementation for that.